### ISSUE TITLE
cuda out of memory

### ISSUE THREAD

**Issue by rl15442** [learnables/learn2learn#161]
when I use Fomaml and Meta-SGD, cuda will be out of memory after few training steps. can you give some advice for this problem?

**Comment by seba-1511** [comment]
Hello @rl15442 ,

Can you provide more information? For example, do you have 4GB+ of CUDA memory? What's the network size you're trying to train? Which dataset are you training on?

Those methods are quite memory-hungry, so it's easy to run out of memory if you're not careful.

**Comment by rl15442** [comment]
Thank you for your response!

the network is GPT2 with 117M size. the data size is 7MB (27971 data totally), as shown in the following picture:  Before the symbol '&', the dialogue act is input, and after the symbol '&', the sentence is ouput. 

the CUDA memory has 16GB memory (tesla P100). the batch size is 8. the meta batch size is 2(2 tasks in a meta batch). the learning rate is 5e-5

if you need more information,  I will share the colab link with you. 


![image](https://user-images.githubusercontent.com/42834831/87587047-7a488f80-c6d9-11ea-9330-415bf9cfce97.png)

**Comment by seba-1511** [comment]
Thanks for the details.

I think your model might be a bit too large: for every adaptation step, you'll need at least two copies of the model parameters (one for parameters, one for gradients). Two things I can suggest: 

a. Reduce the number of adaptation steps. Does it run out of memory with only 1 step?
b. Use FOMAML or Meta-SGD only with the last layer of GPT-2. (i.e. similar to ANIL)

**Comment by rl15442** [comment]
Thank you for your response!

1. I just test it again and I found the meta-SGD still has this problem after runing few steps. (I set fast-adaption-steps(fas)=1 and  taks-per-step(tps)=1)

2. If I set the fast-adaption-steps(fas) to be 1 and taks-per-step(tps) to be 1, is the performance will decrease? 

3. is it possible to only use the last layer of GPT-2  in text generation tasks? is the performance still good?  if my task is like text classification, I think we can use Meta-SGD or ANIL to only train the classification layer.
